Lesson8/src/TextFormatter.java

```diff
@@ -0,0 +1,76 @@
+import java.io.*;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TextFormatter implements MyReader {
+
+    private static final String RESOURCE_DIR = "Lesson8/resources/";
+    private static final String OUTPUT_FILE = RESOURCE_DIR + "new text.txt";
+
+    public void reader(String nameFile) {
+        try (FileReader fileReader = new FileReader(RESOURCE_DIR + nameFile);
+             FileWriter fileWriter = new FileWriter(OUTPUT_FILE, false)) {
+            int i;
+            StringBuilder result = new StringBuilder();
+            char[] text = new char[256];
+            while ((i = fileReader.read(text)) > -1) {
+                if (i < 256) {
+                    text = Arrays.copyOf(text, i);
+                }
+                result.append(text);
+            }
+            String[] line = splitOffers(result.toString());
+            for (int j = 0; j < line.length; j++) {
+                if (counterWord(line[j]) >= 3 && counterWord(line[j]) <= 5) {
+                    fileWriter.write(line[j]);
+                    continue;
+                } else if (checkPalindrome(line[j])) {
+                    fileWriter.write(line[j]);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static String[] splitOffers(String string) {
+        String[] offers = string.split("[.!?]+");
```

Comment:
please rename method and var from offers to sentences.
Text contains sentences

People are getting offers (for a new job for example)